src/SopsSync.ts

```diff
@@ -342,17 +346,82 @@ export class SopsSync extends Construct {
           props.encryptionKey?.grantEncryptDecrypt(provider);
         }
         if (props.parameterNames) {
-          provider.addToRolePolicy(
+          // Avoid too large policies
+          // The maximum size of a managed policy is 6.144 bytes -> 1 character = 1 byte
+          const maxPolicyBytes = 6000; // Keep some bytes as a buffer
+          const arnPrefixBytes = 55; // Content for "arn:aws:ssm:ap-southeast-3:<accountnumer>:parameter/
+          let startAtParameter = 0;
+          let currentPolicyBytes = 300; // Reserve some byte space for basic stuff inside the policy
+          for (let i = 0; i < props.parameterNames.length; i += 1) {
```

Comment:
Could you please refactor this into a dedicated function to maintain readability?

Reply:
Yeah sure, good point. It got larger and larger 🙈